src/ShopPaymentTransactionBundle/objects/TPkgShopPaymentTransactionManagerEndPoint.class.php

```diff
@@ -223,6 +225,60 @@ protected function updateOrderPaidStatusBasedOnCurrentBalance()
         return $bOrderPaidStatusChanged;
     }
 
+    private function updateRealUsedVoucherValueBasedOnTransactions(TdbPkgShopPaymentTransaction $transaction): void
+    {
+        $oTransactionData = $this->getTransactionDataFromOrder(
```

Comment:
The method is really large - could you split it so that each method has at most about 20 lines?

Reply:
done